src/controllers/dynakube/activegate/internal/core/reconciler.go

```diff
@@ -0,0 +1,264 @@
+	"github.com/Dynatrace/dynatrace-operator/src/controllers/dynakube/activegate/capability"
+	"github.com/Dynatrace/dynatrace-operator/src/controllers/dynakube/activegate/secrets"
+	"github.com/Dynatrace/dynatrace-operator/src/controllers/dynakube/activegate/statefulset"
+	"github.com/Dynatrace/dynatrace-operator/src/controllers/dynakube/customproperties"
+	"github.com/Dynatrace/dynatrace-operator/src/kubeobjects"
+	"github.com/Dynatrace/dynatrace-operator/src/kubesystem"
+	"github.com/pkg/errors"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+type Reconciler struct {
+	client.Client
+	Dynakube                         *dynatracev1beta1.DynaKube
+	apiReader                        client.Reader
+	scheme                           *runtime.Scheme
+	feature                          string
+	capabilityName                   string
+	serviceAccountOwner              string
+	capability                       *dynatracev1beta1.CapabilityProperties
+	onAfterStatefulSetCreateListener []statefulset.StatefulSetEvent
+	initContainersTemplates          []corev1.Container
+	containerVolumeMounts            []corev1.VolumeMount
+	volumes                          []corev1.Volume
+}
+
+func NewReconciler(clt client.Client, apiReader client.Reader, scheme *runtime.Scheme,
+	dynakube *dynatracev1beta1.DynaKube, capability capability.Capability) *Reconciler {
+
+	serviceAccountOwner := capability.Config().ServiceAccountOwner
+	if serviceAccountOwner == "" {
+		serviceAccountOwner = capability.ShortName()
+	}
+
+	return &Reconciler{
+		Client:                           clt,
+		apiReader:                        apiReader,
+		scheme:                           scheme,
+		Dynakube:                         dynakube,
+		feature:                          capability.ShortName(),
+		capabilityName:                   capability.ArgName(),
+		serviceAccountOwner:              serviceAccountOwner,
+		capability:                       capability.Properties(),
+		onAfterStatefulSetCreateListener: []statefulset.StatefulSetEvent{},
+		initContainersTemplates:          capability.InitContainersTemplates(),
+		containerVolumeMounts:            capability.ContainerVolumeMounts(),
+		volumes:                          capability.Volumes(),
+	}
+}
+
+func (r *Reconciler) AddOnAfterStatefulSetCreateListener(event statefulset.StatefulSetEvent) {
+	r.onAfterStatefulSetCreateListener = append(r.onAfterStatefulSetCreateListener, event)
+}
+
+func (r *Reconciler) Reconcile() (update bool, err error) {
+	if r.capability.CustomProperties != nil {
+		err = customproperties.NewReconciler(r, r.Dynakube, r.serviceAccountOwner, *r.capability.CustomProperties, r.scheme).
+			Reconcile()
```

Comment:
introduces a dependency on a customproperty reconciler instance which cannot be mocked / faked for testing